tests/gtests/test_SW_Weather.cc

```diff
@@ -686,5 +698,315 @@ namespace {
+           instead of daily read-in values
+         */
+
+         // Initialize any variables
+         int yearIndex = 0, midJanDay = 14;
+         SW_WEATHER *w = &SW_Weather;
+
+         /* Test if monthly values are not being used */
+         SW_WTH_setup();
+
+         // Read in all weather
+         SW_WTH_read();
+
+         // Test the middle of January in year 1980 and see if it's not equal to SW_Sky.r_humidity[0],
+         // SW_Sky.cloudcov[0], and SW_Sky.windspeed[0]
+         // Note: Daily interpolated values in the middle of a month are equal to the
+         // original monthly values from which they were interpolated
+         EXPECT_NEAR(w->allHist[yearIndex]->r_humidity_daily[midJanDay], SW_Sky.r_humidity[0], tol6);
+         EXPECT_NEAR(w->allHist[yearIndex]->cloudcov_daily[midJanDay], SW_Sky.cloudcov[0], tol6);
+         EXPECT_NEAR(w->allHist[yearIndex]->windspeed_daily[midJanDay], SW_Sky.windspeed[0], tol6);
+     }
+
+     TEST(DailyWeatherInputTest, DailyGridMet) {
+
+         /*
+            This section tests humidity-related values that can be averaged,
+            max/min relative humidity, and the calculation of actual vapor pressure,
+            based on the average relative humidity value, is calculated reasonably.
+
+            * This section uses the test directory "*_gridmet".
+          */
+
+         SW_WEATHER *w = &SW_Weather;
+         double result, expectedResult;
+         int yearIndex = 0, year = 1980;
+
+        /* Test correct priority is being given to input values from DAYMET */
+         SW_WTH_setup();
+
+             // Switch directory to gridmet input folder
+         strcpy(w->name_prefix, "Input/data_weather_gridmet/weath");
+
+             // Manually edit index/flag arrays in SW_WEATHER to make test as
+             // realistic as possible
+             // Note: Indices are based on the directory:
+             // Input/data_weather_gridmet/weath.1980
+         w->dailyInputIndices[WIND_SPEED] = 3;
+         w->dailyInputIndices[REL_HUMID_MAX] = 4;
+         w->dailyInputIndices[REL_HUMID_MIN] = 5;
+         w->dailyInputIndices[SHORT_WR] = 6;
+         w->dailyInputFlags[REL_HUMID_MAX] = swTRUE;
+         w->dailyInputFlags[REL_HUMID_MIN] = swTRUE;
+         w->dailyInputFlags[WIND_SPEED] = swTRUE;
+         w->dailyInputFlags[SHORT_WR] = swTRUE;
+         w->n_input_forcings = 7;
+
+             // Using the new inputs folder, read in year = 1980
+         _read_weather_hist(
+             year,
+             w->allHist[0],
+             w->name_prefix,
+             w->n_input_forcings,
+             w->dailyInputIndices,
+             w->dailyInputFlags
+         );
+
+         result = w->allHist[yearIndex]->r_humidity_daily[0];
+
+             // Get expected average from Input/data_weather_gridmet/weath.1980 day 1
+             // hursmax_pct and hursmin_pct
+         expectedResult = (74.17 + 31.42) / 2.;
```

Comment:
here and the other similar expects: it would be great to demonstrate that we have values based on daily inputs and not based on the mean monthly interpolated inputs, e.g.,

```
// Expect that daily relative humidity is derived from hursmax_pct and hursmin_pct
// (and is not interpolated from mean monthly values)
EXPECT_NEAR(
  w->allHist[yearIndex]->r_humidity_daily[midJanDay],
  (74.17 + 31.42) / 2.,
  tol6
);
EXPECT_NE(
  w->allHist[yearIndex]->r_humidity_daily[midJanDay],
  SW_Sky.r_humidity[0],
  tol6
);
```